### PR TITLE
Added remove confirmation pop-up

### PR DESCRIPTION
**Pull request by gedas-k**
Now when you want to delete entries form collection you need to approve it.

### CODE REVIEW

File: src/app/references/remove-confirmation/remove-confirmation.component.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  ngOnInit(): void {
+  }
+
+  onNoClick(): void {
+    this.dialogRef.close();
+  }
+
+  /*removeClick(): void {
+    this.dialogRef.close();
+  }*/
```
Comment (rootis) [inline src/app/references/remove-confirmation/remove-confirmation.component.ts]: Comments should be removed

File: src/app/references/list/list.component.ts [hunk 2]
```diff
@@ -59,7 +61,15 @@ export class ListComponent implements OnInit {
   }
 
   delete(row: object) {
-    this.firebaseService.delete(this.referenceService.document, row);
+    const dialogRef = this.dialog.open(RemoveConfirmationComponent, {
+      width: '500px'
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      if(this.confirmation = result) {
```
Comment (rootis) [inline src/app/references/list/list.component.ts]: a) Not formatted properly
b) It's a wrong expression: if "value assignment succeeds" then ... You don't want to check value assignment. You want to check the result value: if (result) then ...

File: src/app/references/list/list.component.ts [hunk 3]
```diff
@@ -28,6 +29,7 @@ export class ListComponent implements OnInit {
   isTextColumn: boolean;
   isVoteColumn: boolean;
   resources: VoteEntity[];
+  confirmation: boolean;
```
Comment (rootis) [inline src/app/references/list/list.component.ts]: Is it necessary?

File: src/app/references/remove-confirmation/remove-confirmation.component.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<h3 class="mat-h3">Do you  want to remove this record?</h3>
+
+<div mat-dialog-actions>
+  <button mat-button (click)="onNoClick()">Cancel</button>
```
Comment (rootis) [inline src/app/references/remove-confirmation/remove-confirmation.component.html]: Cancel styling should be as secondary button

File: src/app/references/remove-confirmation/remove-confirmation.component.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+  <button mat-button [mat-dialog-close]="true">Remove</button>
```
Comment (rootis) [inline src/app/references/remove-confirmation/remove-confirmation.component.html]: Remove styling should be as primary button

**Comment by gedas-k** [comment]
Made changes according to remarks. Just not sure about not correctly formatted part:

```
    });

    dialogRef.afterClosed().subscribe(result => {
      if(result) {
```

File: src/app/references/list/list.component.ts [hunk 6]
```diff
@@ -59,7 +60,15 @@ export class ListComponent implements OnInit {
...
-    this.firebaseService.delete(this.referenceService.document, row);
+    const dialogRef = this.dialog.open(DeleteConfirmationComponent, {
+      width: '500px'
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      if(result) {
+        this.firebaseService.delete(this.referenceService.document, row);
+      }
+    });
```
Comment (rootis) [inline src/app/references/list/list.component.ts]: this section might be shortened into: result => result && this.firebaseService.delete(this.referenceService.document, row)

Comment (gedas-k) [reply src/app/references/list/list.component.ts]: Done.

**Pull request merged**